### ISSUE TITLE
/usr/lib/libcurl.so.4: version `CURL_OPENSSL_3' not found

### ISSUE THREAD

**Issue by florensie** [blt4linux/blt4l#109]
https://pastebin.com/nBrSL7rq

I have installed blt4l using the AUR package but it could be that it's messed up because I also tried installing and uninstalling a couple of times and different types of installs because it wasn't working. In the end I found out I should add the launch command but then I get this. The output is from running `steam` in the command line and then launching payday.

**Comment by KingBowser** [comment]
In order to obtain suitable debug output from PAYDAY 2, you will need to
launch it from a terminal with a special environment. I will get you
instructions tonight once I am home.

**Comment by RomanHargrave** [comment]
@florensie to elimate the possibility that the game/BLT is faulty, please launch PAYDAY 2 in the following manner


Change your working directory to the PAYDAY 2 install location, and run the following command. This will output the actual game debug logs.

```
env SteamAppId=218620 SteamAppUser=(your steam username) SteamGameId=218620 SteamUser=(your steam username) suppress_restart=1 ./payday2_release
```

**Comment by florensie** [comment]
With those settings the game launches fine with blt working. Can you confirm that the correct launch option for steam is `blt4l %command%`? Also, the game asked me to install beardlib but after the in-game download ended it got stuck on extracting.

https://pastebin.com/2Q53zTr7

**Comment by ZNixian** [comment]
> Can you confirm that the correct launch option for steam is `blt4l %command%`

No, it's `LD_PRELOAD="$LD_PRELOAD libblt_loader.so" %command%`.

After ensuring that `libblt_loader.so` has been copied to your PAYDAY 2 directory.

**Comment by ZNixian** [comment]
Can you check the contents of `mods/logs`, in particular is there a `crash.txt` file and if so, what's it's contents?

**Comment by florensie** [comment]
there is no crash.txt

**Comment by ZNixian** [comment]
That's bizarre. No idea what might be causing it, sorry.

**Comment by florensie** [comment]
When I use /usr/lib/blt4l/libblt_loader.so for LD_PRELOAD it works but I still get the beardlib issue.

**Comment by ZNixian** [comment]
In that case, you probably had an outdated version of BLT in your `PAYDAY 2` directory. No problem if it's working now.

RE BeardLib - that's HoloHUD asking to install it. I'd suggest downloading and installing it manually.

**Comment by RomanHargrave** [comment]
Please try compiling blt4l @florensie

**Comment by florensie** [comment]
@ZNixian Installing manually doesn't seem to work. It still asks to install it when I launch the game and the game crashes.

**Comment by killstrelok** [comment]
I think I've found out the cause of this error.
I'm running Ubuntu 18.04.

So I couldn't get the game working with the command posted above (env SteamAppId=218620 SteamAppUser=(your steam username) SteamGameId=218620 SteamUser=(your steam username) suppress_restart=1 LD_PRELOAD=(path to libblt_loader.so - cannot have spaces) ./payday2_release)
as I'd get the error regarding CURL_OPENSSL_3.

Here's what "fixes" it:
There's a problem with dependencies. Ubuntu 18.04 will refuse to install libcurl3 if libcurl4 is installed. BLT4L will NOT start with libcurl4 installed. To start it, I had to install libcurl3 (I don't recommend doing this, it uninstalled cmake, playonlinux and libcurl4). Then I wouldn't get the error message again.

It's quite a big problem as installing libcurl3, for some strange reason, tanked my FPS by A LOT. We're talking about going down from 100 fps to 10-15. Absolutely crazy. But the mods would work, though.

Is there any chance to make the blt4l depend on libcurl4 instead of 3?

**Comment by dribbleondo** [comment]
These days I have moved onto recommending @ZNixian 's https://superblt.znix.xyz/ as that uses Libcurl4 for pretty much that exact reason.

**Comment by RomanHargrave** [comment]
@dribbleondo @killstrelok I haven't been playing much Payday 2 lately. I have commit access to ZuperBLT, so I may continue work there if I get back in to it - or alternatively may overhall BLT4L.
